analysis/paladinholy/src/modules/items/UntemperedDedication.tsx

```diff
@@ -0,0 +1,111 @@
+  _isFullStack = true;
+  // There is also no buff event generated when LOTM is cast while there are already 5 stacks
+  // so we are maintaining the date of the last refresh here.
+  _lastFullStackRefresh = 0;
+  // a counter for when LOTM stacks are dropped
+  _totalDrops = 0;
+  // tracker for number of current stacks
+  _currentStacks = 5;
+  conduitRank = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasConduitBySpellID(SPELLS.UNTEMPERED_DEDICATION.id);
+    this.conduitRank = this.selectedCombatant.conduitRankBySpellID(SPELLS.UNTEMPERED_DEDICATION.id);
+    if (!this.active) {
+      return;
+    }
+    this.addEventListener(
+      Events.applybuffstack.by(SELECTED_PLAYER).spell(SPELLS.UNTEMPERED_DEDICATION_BUFF),
+      this.onByPlayerUntemperedDedicationBuff,
+    );
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.UNTEMPERED_DEDICATION_BUFF),
+      this.onByPlayerUntemperedDedicationInitialBuff,
+    );
+    this.addEventListener(
+      Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.UNTEMPERED_DEDICATION_BUFF),
+      this.onByPlayerUntemperedDedicationBuffRemoval,
+    );
+  }
+
+  /**
+   * Sets _isFullStack to true if there are now 5 stacks, as well as the last refresh
+   * NB: it comes after the LOTM Cast
+   * @param event indicate the new number of stacks (only positive increments)
+   */
+  onByPlayerUntemperedDedicationBuff(event: ApplyBuffStackEvent) {
+    if (event.stack === 5) {
+      this._isFullStack = true;
+      this._lastFullStackRefresh = event.timestamp;
+    } else {
+      this._isFullStack = false;
+    }
+    this._currentStacks = event.stack;
+  }
+
+  /**
+   * Sets _isFullStack to false.
+   * Note: The only exception is the prepull event, for which we assume that
+   * it means the character has five stacks. It will be corrected if the
+   * first LOTM generates a ApplyBuffStack or ApplyBuff event
+   * Note 2: Technically, the only time this function does something is on the first
+   * apply buff if there was no prepull UD.
+   * @param event with info about first stack, or prepull incomplete indication
+   */
+  onByPlayerUntemperedDedicationInitialBuff(event: ApplyBuffEvent) {
+    if (event.prepull !== true) {
+      this._isFullStack = false;
+      this._currentStacks = 1;
+    }
+  }
+
+  onByPlayerUntemperedDedicationBuffRemoval(event: RemoveBuffEvent) {
+    this._totalDrops += 1;
+    this._isFullStack = false;
+    this._currentStacks = 0;
+  }
+
+  get dropSuggestion() {
+    return {
+      actual: this._totalDrops,
+      isGreaterThan: {
+        minor: 0,
+        average: 1,
+        major: 2,
+      },
+      style: ThresholdStyle.NUMBER,
+    };
+  }
+
+  suggestions(when: When) {
+    when(this.dropSuggestion).addSuggestion((suggest, actual) =>
+      suggest(
+        <span>
+          You dropped your Untempered Dedication stacks {actual} times. Try to keep it up at all
```

Comment:
TECHNICALLY having the stacks up 100% of the time is completely pointless. The only time it matters is if you're gonna LoTM...